RecoHGCal/TICL/plugins/SeedingRegionByTracks.cc

```diff
@@ -0,0 +1,74 @@
+// Author: Arabella Martelli, Felice Pantaleo, Marco Rovere - arabella.martelli@cern.ch, felice.pantaleo@cern.ch, marco.rovere@cern.ch
+// Date: 06/2019
+#include <algorithm>
+#include <set>
+#include <vector>
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "SeedingRegionByTracks.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "TrackingTools/TrajectoryState/interface/TrajectoryStateTransform.h"
+
+using namespace ticl;
+
+SeedingRegionByTracks::SeedingRegionByTracks(const edm::ParameterSet &conf, edm::ConsumesCollector &sumes)
+    : SeedingRegionAlgoBase(conf, sumes),
+      cutTk_(conf.getParameter<std::string>("cutTk")),
+      propName_(conf.getParameter<std::string>("propagator")) {
+  tracks_token_ = sumes.consumes<reco::TrackCollection>(conf.getParameter<edm::InputTag>("tracks"));
+}
+
+SeedingRegionByTracks::~SeedingRegionByTracks(){};
+
+void SeedingRegionByTracks::makeRegions(const edm::Event &ev,
+                                        const edm::EventSetup &es,
+                                        std::vector<ticl::TICLSeedingRegion> &result) {
+  std::call_once(initializeGeometry_, [&]() {
+    edm::ESHandle<HGCalDDDConstants> hdc;
+    es.get<IdealGeometryRecord>().get(detectorName_, hdc);
+    hgcons_ = hdc.product();
+    buildFirstLayers();
+    es.get<IdealMagneticFieldRecord>().get(bfield_);
+    es.get<TrackingComponentsRecord>().get(propName_, propagator_);
+  });
+
+  edm::Handle<reco::TrackCollection> tracks_h;
+  ev.getByToken(tracks_token_, tracks_h);
+  edm::ProductID trkId = tracks_h.id();
+  auto bFieldProd = bfield_.product();
+  const Propagator &prop = (*propagator_);
+
+  int nTracks = tracks_h->size();
+  for (int i = 0; i < nTracks; ++i) {
+    const reco::Track &tk = (*tracks_h)[i];
```

Comment:
for range loop with an index may be better

Reply:
this I don't understand. the "i" is used afterwards so a counter is anyway needed...

Reply:
an index can be followed inside as well; not a strong request anyways